codemods/v4/scripts/default-data-error-value.ts

```diff
@@ -1,22 +1,36 @@
-import type { SgRoot } from "codemod:ast-grep";
+import type { SgRoot, Edit } from "codemod:ast-grep";
 import type TSX from "codemod:ast-grep/langs/tsx";
+import {
+  hasAnyContent,
+  applyEdits,
+  DATA_FETCH_HOOKS,
+  PATTERNS,
+} from "../utils/index.js";
 
-async function transform(root: SgRoot<TSX>): Promise<string> {
+async function transform(root: SgRoot<TSX>): Promise<string | null> {
   const rootNode = root.root();
 
+  // Quick check - does file contain data fetching hooks?
+  if (!hasAnyContent(root, DATA_FETCH_HOOKS)) {
+    return null;
+  }
```

Comment:
We shouldn't encourage string ops, and this is a bad practice imo. This is also a duplicate check. You're already performing an ast-based search later.